smac/intensifier/hyperband_utils.py

```diff
@@ -0,0 +1,184 @@
+    n_configs_in_stage = hyperband_round["n_configs_in_stage"]
+    budgets_in_stage = hyperband_round["budgets_in_stage"]
+
+    remaining_trials = 0
+    for stage in n_configs_in_stage.keys():
+        B = budgets_in_stage[stage]
+        C = n_configs_in_stage[stage]
+        for b, c in zip(B, C):
+            # How many trials are left?
+            # If b * c is lower than remaining budget, we can add full c
+            # otherwise we need to find out how many trials we can do with this budget
+            remaining_trials += min(c, int(np.floor(remaining_budget / b)))
+            # We cannot go lower than 0
+            # If we are in the case of b*c > remaining_budget, we will not have any
+            # budget left. We can not add full c but the number of trials that still fit
+            remaining_budget = max(0, remaining_budget - b * c)
+
+    n_trials = int(number_of_full_hb_rounds * trials_used_per_hb_round + remaining_trials)
+
+    hyperband_info = hyperband_round
+    hyperband_info["n_trials"] = n_trials
+    hyperband_info["total_budget"] = total_budget
+    hyperband_info["eta"] = eta
+    hyperband_info["min_budget"] = min_budget
+    hyperband_info["max_budget"] = max_budget
+
+    return hyperband_info
+
+
+def print_hyperband_summary(hyperband_info: dict) -> None:
+    """Print summary about Hyperband as used in the MultiFidelityFacade
+
+    Parameters
+    ----------
+    hyperband_info : dict
+        Info dict about Hyperband
+    """
+    print("-" * 30, "HYPERBAND IN MULTI-FIDELITY", "-" * 30)
+    print("total budget:\t\t", hyperband_info["total_budget"])
+    print("total number of trials:\t", hyperband_info["n_trials"])
+    print("number of HB rounds:\t", hyperband_info["total_budget"] / hyperband_info["budget_used"])
+    print()
+
+    print("\t~~~~~~~~~~~~HYPERBAND ROUND")
+    print("\teta:\t\t\t\t\t", hyperband_info["eta"])
+    print("\tmin budget per trial:\t\t\t", hyperband_info["min_budget"])
+    print("\tmax budget per trial:\t\t\t", hyperband_info["max_budget"])
+    print("\ttotal number of trials per HB round:\t", hyperband_info["trials_used"])
+    print("\tbudget used per HB round:\t\t", hyperband_info["budget_used"])
+    print("\tnumber of brackets:\t\t\t", hyperband_info["number_of_brackets"])
+    print("\tbudgets per stage:\t\t\t", hyperband_info["budgets_in_stage"])
+    print("\tn configs per stage:\t\t\t", hyperband_info["n_configs_in_stage"])
+    print("-" * (2 * 30 + len("HYPERBAND IN MULTI-FIDELITY") + 2))
+
+
+def get_n_trials_for_hyperband_multifidelity(
+    total_budget: float, min_budget: float, max_budget: float, eta: int = 3, print_summary: bool = True
+) -> int:
+    """Caculate the number of trials needed for multi-fidelity optimization
+
+    Specify the total budget and find out how many trials that equals.
+
+    Parameters
+    ----------
+    total_budget : float
+        Total budget for the complete optimization in fidelity units
```

Comment:
could maybe use more explanation on what is meant by a fidelity unit here, maybe a short example in the description

Reply:
added half sentence